src/python_testing/TC_DISHM_3_1.py

```diff
@@ -0,0 +1,152 @@
+
+    async def send_change_to_mode_cmd(self, newMode) -> Clusters.Objects.DishwasherMode.Commands.ChangeToModeResponse:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.DishwasherMode.Commands.ChangeToMode(newMode=newMode), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.DishwasherMode.Commands.ChangeToModeResponse),
+                            "Unexpected return type for ChangeToMode")
+        return ret
+
+    async def write_on_mode(self, newMode):
+        ret = await self.default_controller.WriteAttribute(self.dut_node_id, [(self.endpoint, Clusters.DishwasherMode.Attributes.OnMode(newMode))])
+        asserts.assert_equal(ret[0].Status, Status.Success, "Writing to OnMode failed")
+
+    async def write_onoff_on_mode(self) -> Clusters.Objects.OnOff.Commands.On:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.OnOff.Commands.On, endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.OnOff.Commands.On),
+                            "Unexpected return type for OnOff")
+
+    async def write_onoff_off_mode(self) -> Clusters.Objects.OnOff.Commands.Off:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.OnOff.Commands.Off, endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.OnOff.Commands.Off),
+                            "Unexpected return type for OnOff")
+
+    @async_test_body
+    async def test_TC_DISHM_3_2(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        self.endpoint = self.matter_test_config.global_test_params['PIXIT_ENDPOINT']
+
+        asserts.assert_true(self.check_pics("DISHM.S.A0000"), "DISHM.S.A0000 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.A0001"), "DISHM.S.A0001 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.A0002"), "DISHM.S.A0002 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.C00.Rsp"), "DISHM.S.C00.Rsp must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.C01.Tx"), "DISHM.S.C01.Tx must be supported")
+
+        attributes = Clusters.DishwasherMode.Attributes
+
+        from enum import Enum
+
+        class CommonCodes(Enum):
+            SUCCESS = 0x00
+            UNSUPPORTED_MODE = 0x01
+            GENERIC_FAILURE = 0x02
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_step(2, "Read OnMode attribute")
+
+        on_mode_dut = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.OnMode)
+
+        logging.info("OnMode: %s" % (on_mode_dut))
+
+        if on_mode_dut == NullValue:
+            self.print_step(2, "Read SupportedModes attribute")
```

Comment:
This does not match the test plan at all, as of test plan SHA de1ca4924fe4d9729ec2295adfeb1589f462924c. Stopping here, because it sounds like I need to be looking at some other set of test plans, but I don't know which ones; PR description does not say.

Reply:
@abpoth Is there an issue tracking TC_RVCRUNM_3_2.py if it's not doing the right thing?